components/InputValidator/InputValidator.jsx

```diff
@@ -0,0 +1,37 @@
+export default function InputValidator(state, action) {
+  if (action.type == "goodCheck") {
+    return {
+      ...state,
+      checking: false,
+      [action.name]: {
+        color: "green",
+        message: action.message,
+        isValid: true,
+      },
+    }
+  } else if (action.type === "badCheck") {
```

Comment:
What is this for? Wouldn't the check be handled when a `success` or `error` happens?

Reply:
Yeah, this is dumb on my part. It should just be called success or error. Sometimes the words just aren't there.